### PR TITLE
Templates

### PR DESCRIPTION
**Pull request by MGatner**
* Adds workflow templates
* Adds config files examples for workflows
* Updates docs with a lot more info

### CODE REVIEW

**Comment by MGatner** [comment]
Hey team, please look over these well as I would like these to trick down to our existing repos.

@samsonasik would you like to take a shot at a suitable Rector version?

**Comment by lonnieezell** [comment]
All in all it looks fine, and I appreciate the updated readme! 

My only question is - do we need Infection to run constantly? I get that it can be a good tool to use when updating your tests, but I feel it introduces a lot of extra noise when checking pull requests. As far as I'm concerned, as long as we make it easy to run manually, it doesn't need to be included on a PR. 

Curious to see what others think, though.

**Comment by MGatner** [comment]
@lonnieezell I picked up on that form our last Infection conversation 🤗 I've actually moved it to its own workflow, which is a change from what is live on `Settings` and `Tasks`. It means that the PHPUnit 8.0 tests run twice but it gives us the option of include/excluding the **infection.yml** workflow in places where we don't want it. I would definitely like to see it included in this repo, as most people have not been exposed to "test quality tools", but I'm fine removing it from existing libraries for now.

**Comment by lonnieezell** [comment]
@MGatner I agree it has a place in this repo. And I'd love to be able to run it manually whenever. Just don't love it when looking over PRs, that's all. :) 

And, to be fair, until you started using it here, I wasn't familiar with it either.

**Comment by MGatner** [comment]
@lonnieezell No worries! Right now there is no template injection or anything, so files supplied here are for developers to copy in as they choose. Eventually I want to include a script to integrate these into repos, especially to merge updates, but for now I suggest that we publish these (once the team has had a chance to review) and then incorporate them manually into our current repos.

**Comment by samsonasik** [comment]
@MGatner for rector, I suggest to use pinned version, eg: 0.12.4:
https://github.com/codeigniter4/CodeIgniter4/blob/d4db7b22dc8fb9901d94ab89c14f84fc662fff0e/composer.json#L27

and use dependabot to verify if it still working with next version.

**Comment by MGatner** [comment]
Thanks @samsonasik. Would the repo's Config file be appropriate as a general configuration for other projects & libraries? Or is there a more generic version we can push?

This also brings up the point: should we delay publishing this repo until PHP 7.4 is the minimum and we can use more specific tooling?

**Review by lonnieezell — APPROVED** [review]

**Comment by samsonasik** [comment]
@MGatner I am not sure, if the "Config" you mean is the "rector.php" configuration, that possibly better to have per-project basis, like in current `CodeIgniter4`'s repo:

https://github.com/codeigniter4/CodeIgniter4/blob/develop/rector.php

File: README.md [hunk 1]
```diff
@@ -5,31 +5,123 @@ Development toolkit for CodeIgniter libraries and projects
...
+
+## Source Files
+
+The provided source files should be considered guidelines or templates for your own use, as
+they may need changing to fit your environment. These are based on the following assumptions:
+
+1. Your default repository branch is set to `develop`
+2. You use Composer to manage all necessary dependencies
+3. Your source code is located in **app/** (for projects) or **src/** (for libraries)
+4. Your unit tests are located in **test/**
```
Comment (kenjis) [inline README.md:51]: `tests`?

Comment (MGatner) [reply README.md:51]: Yes! Good catch.

Comment (MGatner) [inline README.md:51]:
```suggestion
4. Your unit tests are located in **tests/**
```

**Review by kenjis — APPROVED** [review]
I can't see all the details of the workflow settings, but it looks generally good.

**Comment by MGatner** [comment]
@kenjis since these are just recommendations and not usable directly it isn't so important that we get the workflows perfect. That said these are based on my own templates that I've been using and tweaking for many months, took this opportunity to clean them up and consolidate them.

I'm excited to be offering this officially alongside the framework, instead of it always be my own little corner.

**Pull request merged**